src/options.rs

```diff
@@ -0,0 +1,284 @@
+use syn;
+use quote;
+
+#[derive(PartialEq, Debug)]
+pub enum SetterPattern {
+    Owned,
+    Mutable,
+    Immutable
+}
+
+impl Default for SetterPattern {
+    fn default() -> SetterPattern {
+        SetterPattern::Mutable
+    }
+}
+
+#[derive(Debug)]
+pub struct Options {
+    /// e.g. `#[setters]` (defaults to true)
+    setter_enabled: bool,
+    /// e.g. `#[setters(owned)]` (defaults to mutable)
+    setter_pattern: SetterPattern,
+    /// e.g. `#[setters(prefix="with")]` (defaults to None)
+    setter_prefix: Option<String>,
+    /// e.g. `#[setters(private)]` (defaults to public)
+    setter_public: bool,
+    /// e.g. `#[setters(options="implicit")]` (defaults to explicit)
+    setter_implicit_options: bool,
+    /// e.g. `#[getters]` (defaults to false)
+    getter_enabled: bool,
+    /// e.g. `#[getters(prefix="with")]` (defaults to None)
+    getter_prefix: Option<String>,
+    /// e.g. `#[getters(private)]` (defaults to public)
+    getter_public: bool,
+}
+
+impl Options {
+    pub fn setter_pattern(&self) -> &SetterPattern {
+        &self.setter_pattern
+    }
+
+    pub fn setter_visibility(&self) -> Option<quote::Tokens> {
+        if self.setter_public {
+            Some(quote!(pub))
+        } else {
+            None
+        }
+    }
+
+    pub fn setter_enabled(&self) -> bool {
+        self.setter_enabled
+    }
+}
+
+impl<T> From<T> for Options where
+    T: IntoIterator<Item=syn::Attribute>
+{
+    fn from(attributes: T) -> Self {
+        trace!("Parsing struct attributes.");
+        let mut ob = OptionsBuilder::default();
+
+        for attr in attributes {
+            ob.parse_attribute(attr);
+        }
+
+        ob.into()
+    }
+}
+
+#[derive(Default, Debug)]
+pub struct OptionsBuilder {
+    setter_enabled: Option<bool>,
+    setter_pattern: Option<SetterPattern>,
+    setter_prefix: Option<String>,
+    setter_public: Option<bool>,
+    setter_implicit_options: Option<bool>,
+    getter_enabled: Option<bool>,
+    getter_prefix: Option<String>,
+    getter_public: Option<bool>,
+}
+
+impl From<OptionsBuilder> for Options {
+    fn from(b: OptionsBuilder) -> Options {
+        Options {
+            setter_enabled: b.setter_enabled.unwrap_or(true),
+            setter_pattern: b.setter_pattern.unwrap_or_default(),
+            setter_prefix: b.setter_prefix,
+            setter_public: b.setter_public.unwrap_or(true),
+            setter_implicit_options: b.setter_implicit_options.unwrap_or(false),
+            getter_enabled: b.getter_enabled.unwrap_or(false),
+            getter_prefix: b.getter_prefix,
+            getter_public: b.getter_public.unwrap_or(true),
+        }
+    }
+}
+
+impl OptionsBuilder {
```

Comment:
Seriously, the first bunch of methods can be derived, incl. the `warn!` output.

Also just as a warning, every tooling project I work on at some point ends up dogfooding itself :D